packages/nimble-components/src/select/index.ts

```diff
@@ -60,34 +196,203 @@ export class Select extends FoundationSelect implements ErrorPattern {
         this.updateListboxMaxHeightCssVariable();
     }
 
-    // Workaround for https://github.com/microsoft/fast/issues/5773
+    // Workaround for https://github.com/microsoft/fast/issues/5773d
+    // Additionally, we need to force an update to the filteredOptions observable
+    // (by calling 'filterOptions()) so that the template correctly updates.
     public override slottedOptionsChanged(
         prev: Element[],
         next: Element[]
     ): void {
         const value = this.value;
         super.slottedOptionsChanged(prev, next);
+        this.filterOptions();
         if (value) {
             this.value = value;
         }
     }
 
-    private regionChanged(
-        _prev: AnchoredRegion | undefined,
-        _next: AnchoredRegion | undefined
-    ): void {
-        if (this.region && this.control) {
-            this.region.anchorElement = this.control;
+    // disabling linting since the override return type should match the parent class return type
+    // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+    public override clickHandler(e: MouseEvent): boolean | void {
+        const captured = (e.target as HTMLElement).closest<ListOption>(
+            'option,[role=option]'
+        );
+
+        if (!captured?.disabled) {
+            this.updateSelectedIndexFromFilteredSet();
         }
+        return super.clickHandler(e);
     }
 
-    private controlChanged(
-        _prev: HTMLElement | undefined,
-        _next: HTMLElement | undefined
+    public inputClickHandler(e: MouseEvent): void {
+        e.stopPropagation(); // clicking in filter input shouldn't close dropdown
+    }
+
+    /**
+     * Handle content changes on the control input.
+     *
+     * @param e - the input event
+     * @internal
+     */
+    public inputHandler(e: InputEvent): boolean {
+        this.filter = this.filterInputElement?.value ?? '';
+        if (!this.committedSelectedOption) {
+            this.committedSelectedOption = this._options.find(
+                option => option.selected
+            );
+        }
+        this.clearSelection();
+        this.filterOptions();
+
+        if (
+            this.filteredOptions.length > 0
+            && this.committedSelectedOption
+            && !this.filteredOptions.includes(this.committedSelectedOption)
+        ) {
+            this.committedSelectedOption.selected = false;
+            const enabledOptions = this.filteredOptions.filter(
+                o => !o.disabled
+            );
+            if (enabledOptions.length > 0) {
+                enabledOptions[0]!.selected = true;
+            } else {
+                // only filtered option is disabled
+                this.selectedOptions = [];
+                this.selectedIndex = -1;
+                this.filteredOptions[0]!.selected = false;
```

Comment:
Same question here. Am I missing something? Wouldn't `this.filteredOptions[0]!.selected` already be `false` since you called `this.clearSelection()` above in this function?